lib/vanagon/project/dsl.rb

```diff
@@ -159,6 +159,17 @@ def version_from_git
         warn "Directory '#{dirname}' cannot be versioned by git. Maybe it hasn't been tagged yet?"
       end
 
+      def version_from_branch
+        branch = Git.open(File.expand_path("..", Vanagon::Driver.configdir)).current_branch
+        if branch =~ /(\d+(\.\d+)+)/
+          return $1
+        else
+          warn "Can't find a version in your branch, make sure it matches <number>.<number>, like maint/1.7.0/fixing-some-bugs"
```

Comment:
Yes. The test should have good branch names and bad branch names and validate that it doesn't crap out with unexpectedly similar names (thanks to the magic of regex).